servicetalk-concurrent-test-internal/src/main/java/io/servicetalk/concurrent/test/internal/AwaitUtils.java

```diff
@@ -167,14 +167,36 @@ public static boolean awaitUninterruptibly(CountDownLatch latch, long timeout, T
         }
     }
 
+    /**
+     * {@link BlockingQueue#poll(long, TimeUnit)} from the queue, throws unchecked Exception if
+     * thread is interrupted while waiting.
+     * @param queue the queue to poll from.
+     * @param timeout the timeout duration to poll for.
+     * @param unit the units applied to {@code timeout}.
+     * @param <T> The types of objects in the queue.
+     * @return see {@link BlockingQueue#poll(long, TimeUnit)}.
+     */
+    @Nullable
+    public static <T> T poll(BlockingQueue<T> queue, long timeout, TimeUnit unit) {
```

Comment:
I am not a fan of the sneaky-throws but this seems like a reasonable compromise if we really don't want the `throws InterruptedException` on the signature.

Reply:
@bondolo, agreed. We discussed it here: https://github.com/apple/servicetalk/pull/1834#discussion_r712622932